BlueShift-iOS-SDK/InApps/BlueShiftNotificationWebViewController.m

```diff
@@ -266,9 +267,13 @@ - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigati
             [self hideFromWindow:YES];
         } else if([BlueShift sharedInstance].appDelegate.oldDelegate && [[BlueShift sharedInstance].appDelegate.oldDelegate respondsToSelector:@selector(application:openURL:options:)]) {
             if (@available(iOS 9.0, *)) {
-                [[BlueShift sharedInstance].appDelegate.oldDelegate application:[UIApplication sharedApplication] openURL: url options:@{}];
-                [self hideFromWindow:YES];
+                NSDictionary *inAppOptions = [self getInAppOpenURLOptions:nil];
+                [[BlueShift sharedInstance].appDelegate.oldDelegate application:[UIApplication sharedApplication] openURL: url options:inAppOptions];
+                [BlueshiftLog logInfo:[NSString stringWithFormat:@"%@ %@",@"Delivered in-app notification deeplink to AppDelegate openURL method, Deep link - ", [url absoluteString]] withDetails:inAppOptions methodName:nil];
```

Comment:
I see this code block is repeating above in two files. Would it be possible to move them to a helper class?

Reply:
The processing of native(modal and slide in) works differently than the htm in-apps. And that's why deep link is processed differently for native and html in-apps at two places.